core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -417,7 +417,7 @@ private[log] class Cleaner(val id: Int,
     txnIndexFile.delete()
 
     val startOffset = segments.head.baseOffset
-    val records = FileRecords.open(logFile, false, log.initFileSize(), log.config.preallocate)
+    val records = FileRecords.open(logFile, false, log.initFileSize(), false)
```

Comment:
Not exactly. The only difference here is always to create a non-preallocate target cleaned segment files. Most of the times, the cleaned files will not contain too much records, that's why we don't use pre-allocate.